pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -86,7 +85,7 @@ func (l *LibvirtDomainManager) preStartHook(vmi *v1.VirtualMachineInstance, doma
 	// generate cloud-init data
 	cloudInitData := cloudinit.GetCloudInitNoCloudSource(vmi)
 	if cloudInitData != nil {
-		hostName := strings.Split(vmi.Name, ".")[0]
+		hostName := util.SanitizeDNSLabel(vmi.Name)
```

Comment:
Almost there. If you now also move the fallback

```golang
		if vmi.Spec.Hostname != "" {
 			hostName = vmi.Spec.Hostname
 		}
```
into this helper we have the whole hostname logic isolated.